### PR TITLE
make custom fields case sensitive

### PR DESCRIPTION
**Pull request by jwetzell**
- removes the lower casing of custom fields
- doesn't allow duplicate custom fields from CustomFieldForm

### CODE REVIEW

**Comment by cpvalente** [comment]
Thank you for this, the code looks good. 
@alex-Arc , could you please sanity check the implications for the APIs and integrations?

**Review by alex-Arc — APPROVED** [review]

File: apps/client/src/features/app-settings/panel/feature-settings-panel/custom-fields/CustomFieldForm.tsx [hunk 1]
```diff
@@ -66,28 +70,38 @@ export default function CustomFieldForm(props: CustomFieldsFormProps) {
...
-          autoComplete='off'
-        />
+      <div className={style.twoCols}>
+        <div>
+          <Panel.Description>Label (only alphanumeric characters are allowed)</Panel.Description>
+          {errors.label && <Panel.Error>{errors.label.message}</Panel.Error>}
+          <Input
+            {...register('label', {
+              required: { value: true, message: 'Required field' },
+              onChange: () => setValue('key', getValues('label').replaceAll(' ', '_')),
```
Comment (alex-Arc) [inline apps/client/src/features/app-settings/panel/feature-settings-panel/custom-fields/CustomFieldForm.tsx:80]: forgot to use `customFieldLabelToKey` here
```suggestion
              onChange: () => setValue('key', customFieldLabelToKey(getValues('label'))),
```

Comment (jwetzell) [reply apps/client/src/features/app-settings/panel/feature-settings-panel/custom-fields/CustomFieldForm.tsx:80]: That's your own code so I'd say have at it?

Comment (alex-Arc) [reply apps/client/src/features/app-settings/panel/feature-settings-panel/custom-fields/CustomFieldForm.tsx:80]: Yeah I just forgot to use it there 😉 
Just wanted to ask befor I changed stuff you were potentially working on

**Pull request merged**